opensquirrel/ir.py

```diff
@@ -355,7 +363,7 @@ def __init__(
         self.operands = operands
 
     def __repr__(self) -> str:
-        return f"MatrixGate(qubits={self.operands}, matrix={self.matrix})"
+        return f"MatrixGate(qubits={self.operands}, matrix={np.round(self.matrix, self._significant_digits_repr)})"
```

Comment:
Is it wise to represent the matrix of a Matrixgate? If there are many operands involved, this may bloat the log.

Reply:
This could indeed be a large output, but it is mostly for diagnostic purposes, so I would like to provide all the available information to the user.